Gems/Profiler/Code/Source/ImGuiCpuProfiler.cpp

```diff
@@ -790,8 +790,11 @@ namespace Profiler
         const AZStd::sys_time_t deleteBeforeTick = AZStd::GetTimeNowTicks() - frameToFrameTime * m_framesToCollect;
 
         // Remove old frame boundary data
-        auto firstBoundaryToKeepItr = AZStd::upper_bound(m_frameEndTicks.begin(), m_frameEndTicks.end(), deleteBeforeTick);
-        m_frameEndTicks.erase(m_frameEndTicks.begin(), firstBoundaryToKeepItr);
+        if (auto firstBoundaryToKeepItr = AZStd::upper_bound(m_frameEndTicks.begin(), m_frameEndTicks.end(), deleteBeforeTick);
+            firstBoundaryToKeepItr != m_frameEndTicks.begin())
+        {
+            m_frameEndTicks.erase(m_frameEndTicks.begin(), firstBoundaryToKeepItr);
```

Comment:
We should just fix erase to do nothing when the iterators are equal.

Reply:
Yea, @jeremyong-az and I discussed that offline earlier.  Will see about getting that fix up soon.